runtime/index.d.ts

```diff
@@ -35,3 +36,22 @@ declare module "@sapper/service-worker" {
 	export const shell: string[];
 	export const routes: Array<{ pattern: RegExp }>;
 }
+
+declare module "@sapper/common" {
+	export interface PreloadContext {
+		fetch: (url: string, options?: any) => Promise<any>;
+		error: (statusCode: number, message: Error | string) => void;
+		redirect: (statusCode: number, location: string) => void;
+	}
+
+	export interface PreloadPage {
+		host: string;
+		path: string;
+		params: Record<string, string>;
+		query: Record<string, string | string[]>;
+	}
```

Comment:
There's a new `error` property being added in https://github.com/sveltejs/sapper/pull/1498. We should probably go ahead and merge that one and then add it here